src/bot.ts

```diff
@@ -46,9 +46,9 @@ export class BotFactory {
 		bot.on("shardReady", shard => logger.notify(`Shard ${shard} ready`));
 		bot.on("shardReconnecting", shard => logger.info(`Shard ${shard} reconnecting`));
 		bot.on("shardResume", (shard, replayed) => logger.info(`Shard ${shard} resumed: ${replayed} events replayed`));
-		bot.on("shardDisconnect", (event, shard) =>
-			logger.notify(`Shard ${shard} disconnected (${event.code},${event.wasClean}): ${event.reason}`)
-		);
+		bot.on("shardDisconnect", (event, shard) => {
+			logger.notify(`Shard ${shard} disconnected (${event.code},${event.wasClean}): ${event.reason}`);
+		});
```

Comment:
TODO: determine why this is now needed in TypeScript 4.9 and potentially file a bug report with Microsoft

Reply:
The statement in question cannot be a return, or it thinks `event` isn't a `CloseEvent` but a union of every possible first argument to the second argument functions passed to `on`, i.e. every possible first argument to _any_ event handler.

Reply:
https://stackoverflow.com/questions/75050705/why-does-this-code-fail-to-compile-from-typescript-4-7-onward/75051294
microsoft/TypeScript#52152